tests/foreman/cli/test_registration.py

```diff
@@ -295,3 +295,37 @@ def test_positive_custom_facts_for_host_registration(
     for interface in interfaces:
         for interface_name in interface.values():
             assert interface_name in str(host_info['network-interfaces'])
+
+
+@pytest.mark.rhel_ver_list([settings.content_host.default_rhel_version])
+def test_positive_global_registration_with_gpg_repo(
+    module_sca_manifest_org,
+    module_location,
+    module_activation_key,
+    module_target_sat,
+    rhel_contenthost,
+):
+    """Host registration command generated and host is registered successfully with gpg repo enabled.
+
+    :id: 8f01d904-dd52-47eb-b909-975574a7c7c7
+
+    :steps:
+        1. Register host with global registration template with gpg repo and key to Satellite.
+
+    :expectedresults: Host is successfully registered, gpg repo is enabled.
+    """
+    org = module_sca_manifest_org
+    repo_url = settings.repos.gr_yum_repo.url
+    repo_gpg_url = settings.repos.gr_yum_repo.gpg_url
+    result = rhel_contenthost.register(
+        org,
+        module_location,
+        module_activation_key.name,
+        module_target_sat,
+        repo_data=f'repo={repo_url},repo_gpg_key_url={repo_gpg_url}',
+    )
+    assert result.status == 0
+    assert rhel_contenthost.subscribed
+    result = rhel_contenthost.execute('yum -v repolist')
+    assert repo_url in result.stdout
+    assert result.status == 0
```

Comment:
Can we add a additional assertion to install a random package from yum_gpg repo?

Reply:
The rpms available in yum_gpg repo are not signed.

Reply:
I tried that day as we were discussing, but it seems the gpg key in the repo is not the key used to sign those rpms in the repo.

Reply:
Created a task to sign all the packages present in yum_gpg repo with the gpg key present: https://issues.redhat.com/browse/SAT-27653